src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -48,16 +52,24 @@ private AutoSetFileLinksUtil(List<Path> directories, AutoLinkPreferences autoLin
 
     public List<BibEntry> linkAssociatedFiles(List<BibEntry> entries, NamedCompound ce) {
         List<BibEntry> changedEntries = new ArrayList<>();
+
+        List<IOException> fileExceptions = new ArrayList<>(); //change
         for (BibEntry entry : entries) {
 
             List<LinkedFile> linkedFiles = new ArrayList<>();
             try {
                 linkedFiles = findAssociatedNotLinkedFiles(entry);
             } catch (IOException e) {
+                fileExceptions.add(e); //change
                 LOGGER.error("Problem finding files", e);
+
             }
 
             if (ce != null) {
+                if (!fileExceptions.isEmpty()) { // is this where it goes? find out!
```

Comment:
I would propose to move this into the succeeded piece of code:
https://github.com/JabRef/jabref/blob/5ab494e8b1732f688d4ed8d696740bc7e8ef4101/src/main/java/org/jabref/gui/externalfiles/AutoLinkFilesAction.java#L60-L68

Reply:
Oh ok, thank you. Can I ask you a few questions about the `succeeded()` method?

As far as I understood, it notifies the user when it has set the file links, i.e. when the returned List of linkAssociatedFiles(...) is not empty. In the other case `Finished automatically setting external links. No files found.`  I haven't quite understood yet, how this `else` case is different from what I am trying to accomplish.

Reply:
Oh ok, thank you. Can I ask you a few questions about the `succeeded()` method?

As far as I understood, it notifies the user when it has set the file links, i.e. when the returned List of linkAssociatedFiles(...) is not empty. In the other case `Finished automatically setting external links. No files found.`  I haven't quite understood yet, how this `else` case is different from what I am trying to accomplish.

Reply:
Sorry for the delay in answering. The succeeded method is called whenever the task is finished normally and doesn't get interrupted. 
It either says files found or no files found, but it doesn't tell the user if there was an error (exception) which prevented  finding the files. 
So one addtional else case (your changes) is to notify the users that there have been X exceptions with the hint 
e.g. There were x files with an exception. See log for details